### PR TITLE
Initial implementation

### PR DESCRIPTION
**Pull request by sayakpaul**

### CODE REVIEW

File: prepare_jsonl_shards.py [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+        os.makedirs(args.local_dir)
+
+    wikitext = datasets.load_dataset(
+        "wikitext", "wikitext-103-raw-v1", split="train"
+    )
+
+    if args.limit is not None:
+        wikitext = wikitext[: args.limit]
+        wikitext = Dataset.from_dict(wikitext)
+        logging.info(f"Limiting the dataset to {args.limit} entries.")
```
Comment (lhoestq) [inline prepare_jsonl_shards.py]: `wikitext = wikitext[: args.limit]` brings the whole subset in memory, it is more efficient to just load the requested subset in `load_dataset`directly:

```suggestion
    if args.limit is None:
        wikitext = datasets.load_dataset(
            "wikitext", "wikitext-103-raw-v1", split="train"
        )
    else:
        logging.info(f"Limiting the dataset to {args.limit} entries.")
        wikitext = datasets.load_dataset(
            "wikitext", "wikitext-103-raw-v1", split=f"train[:{args.limit}]"
        )
```

File: requirements.txt [hunk 2]
```diff
@@ -0,0 +1,6 @@
+apache-beam[gcp]==2.34.0
+torch==1.11.0
+tokenizers==0.11.6
+transformers==4.17.0
+datasets==2.0.0
```
Comment (lhoestq) [inline requirements.txt]: any reason to pin `datasets==2.0.0` ? The latest version is `2.2.2`

**Review by lhoestq — COMMENTED** [review]
Nice ! just one nit:

File: requirements.txt (hunk 2 above)
Comment (sayakpaul) [reply requirements.txt]: Nothing particular. We have been using 2.0.0 at Carted for some time. That's why. Can bump it up.

**Comment by sayakpaul** [comment]
Thanks @lhoestq!

**Pull request merged**